### PR TITLE
Always deploy SNAPSHOT when main branch is updated

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

I've split the Build and Test workflow from the deployment workflow.

Build and Test gets run when a PR is
- opened
- reopened
- synchronised (new changes were added).

Deploy gets run when 
- there are new changes on `main` (deploys SNAPSHOT)
- when it is triggered manually (deploys SNAPSHOT)
- when a new release is published (deploy actual version).

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to backport the fix

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually

Documentation:
* [ ] Javadoc has been written
* [ ] The documentation is updated

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up Java environment
+        uses: actions/setup-java@v2.4.0
+        with:
+          distribution: temurin
+          java-version: 11
+          gpg-private-key: ${{ secrets.MAVEN_CENTRAL_GPG_SIGNING_KEY_SEC }}
```
Comment (pihme) [inline .github/workflows/build-test.yml:25]: :question: Why do we need the Maven central stuff if we don't publish anything in this build?

**Review by pihme — COMMENTED** [review]
I don't get the change, tbh.

The deploy artifacts job seems to not build the artifact. Maybe it takes the artifact from the other workflow run, but I don't see how they are sequenced.

Could you explain how this is supposed to work?

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (remcowesterhoud) [reply .github/workflows/build-test.yml:25]: That's a good question. I just copied it from what we already had, but I'll remove it.

**Comment by remcowesterhoud** [comment]
@pihme It uses the [community action snapshot release](https://github.com/camunda-community-hub/community-action-maven-release/blob/main/action.yml). This builds the artifact for us:

```yaml
    - name: Publish SNAPSHOT
      run: |-
        test -n "${{ inputs.release-version }}" && echo "::debug::Not publishing SNAPSHOT because release-version is set" && exit 0
        mvn -B ${{ inputs.maven-additional-options }} -DskipTests ${{ inputs.maven-release-options }} deploy
        find . -path '**target/nexus-staging/deferred/.index' -exec sed -i 's@:camunda-nexus:.*$@:central:https://${{inputs.maven-url}}/content/repositories/snapshots/@g' {} +
        mvn -B ${{ inputs.maven-additional-options }} -DskipTests ${{ inputs.maven-release-options }} org.sonatype.plugins:nexus-staging-maven-plugin:deploy-staged
      shell: bash
      env:
        NEXUS_USR: ${{ inputs.nexus-usr}}
        NEXUS_PSW: ${{ inputs.nexus-psw }}
        MAVEN_USR: ${{ inputs.maven-usr }}
        MAVEN_PSW: ${{ inputs.maven-psw }}
    - name: Publish Maven Release
      run: |-
        test -z "${{ inputs.release-version }}" && echo "::debug::Skipping Release because release-version is unset" && exit 0
        echo "Set version to ${{ inputs.release-version }}"
        mvn -B ${{ inputs.maven-additional-options }} versions:set org.codehaus.mojo:versions-maven-plugin:2.8.1:update-child-modules -DnewVersion=${{ inputs.release-version }}
        echo "Deploy release to Camunda Repository using ${{inputs.release-profile}} profile"
        mvn -B ${RELEASE_PROFILE} ${{ inputs.maven-additional-options }} -DskipTests ${{ inputs.maven-release-options }} deploy
        echo "Deply release to Maven Central using profiles ${{inputs.release-profile}}, ${{inputs.central-release-profile}}"
        mvn -B ${RELEASE_PROFILE} ${CENTRAL_RELEASE_PROFILE} ${{ inputs.maven-additional-options }} -DskipTests -DnexusUrl=https://${{inputs.maven-url}}/  ${{ inputs.maven-release-options }} deploy
      shell: bash
      env:
        NEXUS_USR: ${{ inputs.nexus-usr}}
        NEXUS_PSW: ${{ inputs.nexus-psw }}
        MAVEN_USR: ${{ inputs.maven-usr }}
        MAVEN_PSW: ${{ inputs.maven-psw }}
        MAVEN_GPG_PASSPHRASE: ${{ inputs.maven-gpg-passphrase }}
```

The workflows are not chained together at all. The first one only runs on pull-requests, the deploy one only runs when new changes are added to the `main` branch or when a release is published.

**Review by pihme — APPROVED** [review]
Ok, thanks for the explanation!

**Pull request merged**